flask/app/variants.py

```diff
@@ -227,17 +226,108 @@ def parse_gene_panel() -> List[Any]:
         app.logger.error("Not all requested genes were found.")
         abort(400, description="Not all requested genes were found.")
 
-    genes = query.all()
+    # genes_list = query.all()
+
+    return gene_filter, found_genes
+
+
+def parse_region(region: str):
+    """
+    Parses a given list of ranges (eg. "chr1:0500-0509,chr5:0600-0630,chr18:0440-0330").
+    Returns a filter for use against the variant table to find corresponding variants
+    based on gene start and end positions.
+    """
+    CHR, START, END = 0, 1, 2
+    # format check
+    # 1: chromosome, 2: start position, 3: end position
+    region_pattern = re.compile("chr([\da-zA-Z]+):([\d]+)-([\d]+)")
+    matches = [region_pattern.match(r) for r in region.split(",")]
+
+    app.logger.info("Requested region: %s", region)
+
+    if not all(match is not None and len(match.groups()) == 3 for match in matches):
+        app.logger.error("Invalid region format: %s", region)
+        abort(400, description="Invalid region format")
+
+    # prepare set of regions
+    region_set = set([match.groups() for match in matches])
+    for r in region_set:
```

Comment:
It looks like the set itself isn't being modified with the corrected order, so the set that's iterated over to generate the filters isn't actually corrected. A solution might be to convert the set to a list and index into the list for assignment eg. for i, r in enumerate(region_list)